### ISSUE TITLE
JIT decompose pass: To enable decomposition only for GPU device

### ISSUE THREAD

**Issue by arthuryuan1987** [pytorch/pytorch#73639]
### 🐛 Describe the bug

The purpose is to enable decomposition pass only for GPU device, here listed two reasons,
1. On our device backend, we hope to have an operator level optimization for BatchNorm and LayerNorm (mkldnn support) on JIT graph. So we don't want to decompose these two operators.
2. According to the current implementation and my understanding, decomposition pass only benefits GPU device, where we can get elementwise kernels fusion by TensorExpr lowering. But the condition is to enable decomposition for all non-CPU devices,
```
 44   auto device = input->type()->expectRef<TensorType>().device();
 45   // As of now, we do the decomposition for batchnorm/layernorm on GPU device
 46   // only
 47   if (!device || (*device).is_cpu()) {
 48     return false;
 49   }
```

### Versions

master

**Comment by arthuryuan1987** [comment]
#https://github.com/pytorch/pytorch/pull/73637

**Comment by davidberard98** [comment]
closed by #73637
